### ISSUE TITLE
`no-useless-flag` false positive for `i` flag and Unicode property escapes

### ISSUE THREAD

**Issue by RunDevelopment** [ota-meshi/eslint-plugin-regexp#265]
See the following example:

```js
var re1 = /\p{Ll}/iu;
```

![image](https://user-images.githubusercontent.com/20878432/124462849-5bcc2000-dd92-11eb-88f2-28220ade0ef9.png)

This report is incorrect as `\p{Ll}` matches all lower-case letters. The `i` flag is very much necessary.

**Comment by RunDevelopment** [comment]
This is most likely caused by an upstream bug in refa.

I just fixed that Unicode property escapes were incorrectly canonicalized. This was incorrect because the canonicalization is kinda counter-intuitive. I'll try to release a new version of refa by the end of the week.
